pkg/getter/httpgetter.go

```diff
@@ -82,21 +82,21 @@ func (g *HTTPGetter) get(href string) (*bytes.Buffer, error) {
 }
 
 // newHTTPGetter constructs a valid http/https client as Getter
-func newHTTPGetter(url, certFile, keyFile, caFile string) (Getter, error) {
-	return NewHTTPGetter(url, certFile, keyFile, caFile)
+func newHTTPGetter(options Options) (Getter, error) {
```

Comment:
This may be out of scope for this PR, but is there a reason the exported `NewHTTPGetter` doesn't just return a `Getter`? Based on what I am seeing in these files, we no longer need to export the `HTTPGetter` and just hide it behind the `Getter` interface. The `HTTPGetter` specific functions are now covered by the `Options` config

Reply:
Some parts of the codebase still rely on NewHTTPGetter returning an HTTPGetter so it can call SetBasicAuth (and SetUserAgent, though that isn't used in the codebase). We can certainly do that in a follow-up now that these options are being passed in and set when the getter is being instantiated.